components/I18nAddressFields.js

```diff
@@ -0,0 +1,213 @@
+  const zones = get(addressInfo, 'zones', []);
+  if (mappedMatches.includes('zone') && !isEmpty(zones)) {
+    const zoneIndex = addressFormFields.find(idx => idx[0] === 'zone');
+    zoneIndex.push(addressInfo.zones);
+  }
+
+  return addressFormFields;
+};
+
+/** Checks if the key from labels is also present in optionalLabels */
+const isFieldOptional = (addressInfo, fieldName) => {
+  return !Object.keys(addressInfo.optionalLabels).includes(fieldName);
+};
+
+/** Controlled component to be used in forms that require addresses that need some validation
+ * i.e. Expenses and Contributions. Can be used with or without Formik.
+ * See PropTypes info for differing props based on Formik vs. non-Formik.*/
+const I18nAddressFields = ({ selectedCountry, value, onChange }) => {
+  const intl = useIntl();
+
+  /** Pass user's chosen locale to AddressFormatter if present. */
+  if (intl.locale) {
+    addressFormatter.updateLocale(intl.locale);
+  }
+
+  /** If country chosen from InputTypeCountry is one of missingCountries, use 'US' instead */
+  const country = missingCountries.includes(selectedCountry) ? 'US' : selectedCountry;
+
+  /** Prepare the address form data */
+  const [data, setData] = React.useState(null);
+  const [fields, setFields] = React.useState(null);
+  const [loading, setLoading] = React.useState(true);
+
+  React.useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const response = await addressFormatter.getCountry(country);
+        setData(pick(response, ['formatting', 'labels', 'optionalLabels', 'zones']));
+        const countryInfo = pick(response, ['formatting', 'labels', 'optionalLabels', 'zones']);
+        const addressFields = wrangleAddressData(countryInfo);
+        setFields(addressFields);
+      } catch (e) {
+        console.warn('Call to Shopify API failed. Falling back to plain address form. Error: ', e.message);
+      } finally {
+        setLoading(false);
+      }
+    };
+
+    fetchData();
+  }, [selectedCountry]);
+
+  if (loading) {
+    return <LoadingPlaceholder width="100%" height={163} />;
+  }
+
+  // If the call to Shopify fails, fallback to the free text address field
+  if (!loading && (!data || !fields)) {
+    return (
+      <StyledInputField name="payeeLocation.address" label="address" labelFontSize="13px" required mt={3}>
+        {inputProps => (
+          <StyledTextarea
+            {...inputProps}
+            onChange={e => {
+              onChange(e.target);
+            }}
+            value={typeof value === 'object' ? JSON.stringify(value) : value}
+            minHeight={100}
+            placeholder="P. Sherman 42&#10;Wallaby Way&#10;Sydney"
+          />
+        )}
+      </StyledInputField>
+    );
+  }
+
+  // Zone select
+  const getZoneLabel = zone => {
+    return `${truncate(zone.name, { length: 30 })} - ${zone.code}`;
+  };
+
+  const getZoneSelectOptions = memoizeOne(zones => {
```

Comment:
Memoization here has no effect, `memoizeOne` will just return a new function for each render.

To make it work, you would have to wrap the callback passed to `memoizeOne` in `useCallback`.
But the simplest approach in functional components is to directly use `useMemo`.